[BUG] bloop compile times out on Databricks 11.3

**Describe the bug**
VS Code / Metals/ Bloop IDE for is broken for Shim Development on Databricks 11.3.

The visible symptom is that compile sessions time out and restart after 30 seconds of failing to report the progress.

To see the root cause you need to install bloop CLI https://github.com/gerashegalov/rapids-shell/blob/master/src/bash/init-databricks-shimdev.sh#L11

```
bloop compile --verbose=true rapids-4-spark-sql_2.12
```

which fails in the background with

```
Caused by: java.lang.ClassNotFoundException: org.apache.logging.log4j.core.Filter
        at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:419)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:352)
        ... 52 more
```

which we have seen in #7098, but  the compile keeps blocking without exiting, which looks like a bug in Bloop

--- a/jenkins/databricks/build.sh
+++ b/jenkins/databricks/build.sh
@@ -285,9 +285,12 @@ set_dep_jars()
     # Delta Lake support is included in SQL jar
     artifacts[DELTA]="-DgroupId=io.delta -DartifactId=delta-core_${SCALA_VERSION}"
     dep_jars[DELTA]=${dep_jars[SQL]}
+
     # log4j-core
-    artifacts[LOG4JCORE]="-DgroupId=org.apache.logging.log4j -DartifactId=log4j-core"
-    dep_jars[LOG4JCORE]=${PREFIX_WS_SP_MVN_HADOOP}--org.apache.logging.log4j--log4j-core--org.apache.logging.log4j__log4j-core__${sw_versions[LOG4JCORE]}.jar
+    if [[ "$BASE_SPARK_VERSION" == "3.3.0" ]]; then
+        artifacts[LOG4JCORE]="-DgroupId=org.apache.logging.log4j -DartifactId=log4j-core"
+        dep_jars[LOG4JCORE]=${PREFIX_WS_SP_MVN_HADOOP}--org.apache.logging.log4j--log4j-core--org.apache.logging.log4j__log4j-core__${sw_versions[LOG4JCORE]}.jar
+    fi
 
     # spark-3.1.2 overrides some jar naming conventions
     if [[ $BASE_SPARK_VERSION == "3.1.2" ]]

--- a/jenkins/databricks/build.sh
+++ b/jenkins/databricks/build.sh
@@ -166,6 +166,7 @@ set_sw_versions()
             sw_versions[ORC]="1.7.6"
             sw_versions[PARQUET]="1.12.0"
             sw_versions[PROTOBUF]="2.6.1"
+            sw_versions[LOG4JCORE]="2.18.0"
             ;;
         "3.2.1")
             sw_versions[ARROW]="2.0.0"
@@ -284,6 +285,9 @@ set_dep_jars()
     # Delta Lake support is included in SQL jar
     artifacts[DELTA]="-DgroupId=io.delta -DartifactId=delta-core_${SCALA_VERSION}"
     dep_jars[DELTA]=${dep_jars[SQL]}
+    # log4j-core
+    artifacts[LOG4JCORE]="-DgroupId=org.apache.logging.log4j -DartifactId=log4j-core"
+    dep_jars[LOG4JCORE]=${PREFIX_WS_SP_MVN_HADOOP}--org.apache.logging.log4j--log4j-core--org.apache.logging.log4j__log4j-core__${sw_versions[LOG4JCORE]}.jar
 
     # spark-3.1.2 overrides some jar naming conventions
     if [[ $BASE_SPARK_VERSION == "3.1.2" ]]
@@ -328,6 +332,11 @@ else
     echo "!!!! Installing dependendecies. Set SKIP_DEP_INSTALL=1 to speed up reruns of build.sh"# Install required dependencies.
     install_dependencies
 fi
+
+if [[ "$WITH_BLOOP" == "1" ]]; then
+    MVN_OPT="ch.epfl.scala:maven-bloop_2.13:bloopInstall $MVN_OPT"
+fi
+
 # Build the RAPIDS plugin by running package command for databricks
 mvn -B -Ddatabricks -Dbuildver=$BUILDVER clean package -DskipTests $MVN_OPT
 

--- a/sql-plugin/pom.xml
+++ b/sql-plugin/pom.xml
@@ -205,6 +205,23 @@
             </dependencies>
         </profile>
         <profile>
+            <id>release330db</id>
+            <activation>
+                <property>
+                    <name>buildver</name>
+                    <value>330db</value>
+                </property>
+            </activation>
+            <dependencies>
+                <dependency>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-core</artifactId>
+                    <version>${spark.version}</version>
+                    <scope>provided</scope>
+                </dependency>
+            </dependencies>
+        </profile>
+        <profile>
             <!--
                  Note that we are using the Spark version for all of the Databricks dependencies as well.
                  The jenkins/databricks/build.sh script handles installing the jars as maven artifacts.
@@ -415,12 +432,6 @@
                 <dependency>
                     <groupId>com.google.protobuf</groupId>
                     <artifactId>protobuf-java</artifactId>
-                    <version>${spark.version}</version>
-                    <scope>provided</scope>
-                </dependency>
-                <dependency>
-                    <groupId>org.apache.logging.log4j</groupId>
-                    <artifactId>log4j-core</artifactId>
                     <version>${spark.version}</version>
                     <scope>provided</scope>
                 </dependency>

--- a/sql-plugin/pom.xml
+++ b/sql-plugin/pom.xml
@@ -418,6 +418,12 @@
                     <version>${spark.version}</version>
                     <scope>provided</scope>
                 </dependency>
+                <dependency>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-core</artifactId>
+                    <version>${spark.version}</version>
+                    <scope>provided</scope>
+                </dependency>
             </dependencies>
         </profile>
 

